code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -154,6 +164,31 @@ LINEN BINS
 
 
 /obj/structure/bedsheetbin/attackby(obj/item/I as obj, mob/user as mob)
+	if(iswrench(I))
+		wrenchAnchor(user, time_to_wrench = 2 SECONDS)
+		return
+	if(iswelder(I))
+		if(anchored)
+			to_chat(user, "<span class='warning'>\The [src] is still secured to whatever surface it is on. Unsecure it first!</span>")
+			return
+		var/obj/item/weapon/weldingtool/W = I
+		if(W.remove_fuel(2,user))
+			to_chat(user, "<span class='notice'>You break \the [src] down into a pile of rods.</span>")
+			new /obj/item/stack/rods(get_turf(src),rand(3,5))
+			if(sheets)
```

Comment:
`sheets` is always a list, this shouldn't be necessary

Reply:
Oh, sheets.len, sorry

Reply:
if `sheets.len` is 0 the loop will simply not run, either way